src/sinks/socket.rs

```diff
@@ -164,63 +159,97 @@ mod test {
     // This is a test that checks that we properly receieve all events in the
     // case of a proper server side write side shutdown.
     //
-    // This test basically sends 10 events shutsdown the server and forces a
+    // This test basically sends 10 events, shutsdown the server and forces a
     // reconnect. It then forces another 10 events through and we should get a
     // total of 20 events.
     //
     // If this test hangs that means somewhere we are not collecting the correct
     // events.
+    #[cfg(feature = "sources-tls")]
     #[test]
     fn tcp_stream_detects_disconnect() {
+        use crate::tls::{MaybeTlsSettings, TlsConfig, TlsOptions};
+        use futures01::{Async, Stream};
+        use std::io::{ErrorKind, Read};
+        use std::sync::{
+            atomic::{AtomicUsize, Ordering},
+            Arc,
+        };
+        use tokio01::io::AsyncWrite;
+
         let addr = next_addr();
         let config = SocketSinkConfig {
             mode: Mode::Tcp(TcpSinkConfig {
                 address: addr.to_string(),
                 encoding: Encoding::Text.into(),
-                // TODO: enable TLS here since this is where
-                // ran into the issue before!
-                tls: None,
+                tls: Some(TlsConfig {
+                    enabled: Some(true),
+                    options: TlsOptions {
+                        verify_certificate: Some(false),
```

Comment:
It would probably be good to enable this as well if we can to ensure we test all realistic code paths.

Reply:
It's not possible enable it while self-signed certs used (`localhost.crt`).